tests/testswitch.nim

```diff
@@ -713,7 +713,9 @@ suite "Switch":
       readers.add(closeReader())
 
     await allFuturesThrowing(readers)
-    await switch2.stop() #Otherwise this leeks
+    await switch2.stop() #Otherwise this leaks
+    await sleepAsync(500.millis)
```

Comment:
We can avoid slowing down the test run by waiting for the connection to drop:

```suggestion
    check await checkExpiring(not switch1.isConnected(switch2.peerInfo.peerID))
```

Tested here: https://github.com/status-im/nim-libp2p/runs/2892600589?check_suite_focus=true

Reply:
:+1: